open-craft/openedx-completion-aggregator
# [BB-6087] feat: exclude `StartDateTransformer` usage from `init_course_blocks`

**Description:**

Excludes `StartDateTransformer` from the transformers list in `init_course_blocks` function,
so all hidden course content is taken into account when calculating course completion.

For example, if a course has already started, and all units are completed, except one, start date
for which is in the future, the completion won't be 100%.

**JIRA:**
- [BB-6087](https://tasks.opencraft.com/browse/BB-6087)

**Testing instructions:**

Prerequisites:
1. Install Completion Aggregator and switch to `0x29a/bb6087/count_hidden_content` branch.
2. Enable `completion.enable_completion_tracking` waffle switch.
3. Set  `COMPLETION_AGGREGATOR_AGGREGATE_UNRELEASED_BLOCKS` Django setting to `True`.

Testing steps:
1. Create a course.
2. Add two sections.
4. Add and publish a unit with `Checkbox` problem to each section.
5. Go to `Settings` -> `Schedule & Details`, and set `Course Start Date` to `01/01/2021`.
6. Go back the the course edit page, and change the release date of the second unit to `01/01/2030`.
7. Create a new user, and enroll into the course. You should see only one section. Complete the unit from it.
8. Run `python manage.py lms run_aggregator_service`.
9.  Open `<LMS>/completion-aggregator/v1/course/<COURSE_ID>/` as a staff user.
You should see, that your regular user has completed **50%** of the course.
